src/DotNetProjectFile.Analyzers/Analyzers/MsBuild/UseCoverletCollectorOrMsBuild.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace DotNetProjectFile.Analyzers.MsBuild;
+
+/// <summary>Implements  <see cref="Rule.UseCoverletCollectorOrMsBuild"/>.</summary>
+[DiagnosticAnalyzer(LanguageNames.CSharp, LanguageNames.VisualBasic)]
+public sealed class UseCoverletCollectorOrMsBuild()
+    : MsBuildProjectFileAnalyzer(Rule.UseCoverletCollectorOrMsBuild)
+{
+    /// <inheritdoc />
+    public override IReadOnlyCollection<ProjectFileType> ApplicableTo => ProjectFileTypes.ProjectFile;
+
+    /// <inheritdoc />
+    protected override void Register(ProjectFileAnalysisContext<MsBuildProject> context)
+    {
+        var collector = context.File
+            .Walk()
+            .OfType<PackageReference>()
+            .LastOrDefault(r => r.Include == NuGet.Packages.coverlet_collector.Name);
```

Comment:
Why last and not first?

Reply:
Biggest change to get it in the project file.